packages/viz/src/providers/Provider.tsx

```diff
@@ -0,0 +1,151 @@
+import { HvTheme, useTheme } from "@hitachivantara/uikit-react-core";
+import * as echarts from "echarts/core";
+import { createContext, useEffect, useState } from "react";
+
+export interface HvVizContextValue {
+  /**
+   * Current theme
+   */
+  theme?: string;
+}
+
+export const HvVizContext = createContext<HvVizContextValue>({
+  theme: undefined,
+});
+
+export interface HvVizProviderProps {
+  /**
+   * Component tree.
+   */
+  children?: React.ReactNode;
+}
+
+/**
+ * Enables theming capabilities for visualizations.
+ *
+ * Without this provider the visualizations will not comply to the UI Kit themes.
+ *
+ * This provider should always be used in combination with the `HvProvider` from
+ * the core package since the former uses the themes provided by the latter.
+ *
+ * `HvVizProvider` should always be used after `HvProvider` like so to work properly:
+ *
+ * ```
+ * <HvProvider>
+ *    <HvVizProvider>
+ *        (...)
+ *    </HvVizProvider>
+ * </HvProvider>
+ * ```
+ */
+export const HvVizProvider = ({ children }: HvVizProviderProps) => {
+  const { activeTheme, colorModes, selectedMode, selectedTheme } = useTheme();
+
+  const [theme, setTheme] = useState<string | undefined>();
+
+  const registerThemes = (
+    themeName: string,
+    modes: string[],
+    themeStructure?: HvTheme
```

Comment:
This looks like a pure function - it could be extracted from the component

Reply:
I will leave it there for now, but yes this is something we can extract later on 🙏